packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -83,12 +89,28 @@ contract OptimismPortal is ResourceMetering {
     mapping(bytes32 => bool) public finalizedWithdrawals;
 
     /**
-     * @param _l2Oracle                  Address of the L2OutputOracle.
-     * @param _finalizationPeriodSeconds Finalization time in seconds.
+     * @notice Reserve extra slots (to to a total of 50) in the storage layout for future upgrades.
+     */
+    uint256[48] private __gap;
+
+    /**
+     * @notice The constructor sets immutable values in the implementation.
+     *         This means that these values can only changed by an upgrade. But the efficiency gains
+     *         are worthwhile.
+     *         Also ensures that the implementation is initialized upon deployment.
      */
     constructor(L2OutputOracle _l2Oracle, uint256 _finalizationPeriodSeconds) {
```

Comment:
These constructor args need to be set via `initialize`

Reply:
They are immutable, so they have to be in the constructor. I verified there are [no gotchas](https://forum.openzeppelin.com/t/un-safe-usage-of-immutable-in-uups/28783/2?u=maurelian) with this approach, and these values should not be expected to change so often that they can't be modified via an upgrade.

https://github.com/ethereum-optimism/optimism/blob/m/upgradeable-portal/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol#L62-L67